packages/views.client-react.unstable/src/context.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2024 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software=
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type {
+  AsyncParameterValueMap,
+  IFoundryViewClient,
+  ParameterConfig,
+} from "@osdk/views-client.unstable";
+import React, { useContext } from "react";
+
+export interface FoundryViewClientContext<CONFIG extends ParameterConfig> {
+  emitEvent: IFoundryViewClient<CONFIG>["emit"];
+  sendReady: IFoundryViewClient<CONFIG>["ready"];
+  parameterValues: AsyncParameterValueMap<CONFIG>;
+}
+
+export const FoundryViewContext = React.createContext<
+  FoundryViewClientContext<ParameterConfig>
+>({
+  emitEvent: () => {},
+  sendReady: () => {},
+  parameterValues: {},
+});
+
+/**
+ * @returns The current FoundryViewClientContext, in the context of your specific parameter configuration
+ */
+export function useFoundryViewContext<CONFIG extends ParameterConfig>() {
+  return useContext(FoundryViewContext) as FoundryViewClientContext<CONFIG>;
```

Comment:
If you want "perfection" you would not use `useFoundryViewContext` nor `FoundryViewContext` and instead could do something like this in the consumer's code

```ts
const { FoundryViewContext, useFoundryViewContext } = createFoundryViewReact<MyConfig>();
```

It requires the extra step of indirection but guarantees that you cannot screw up when doing `useFoundryViewContext()` as you would not need to pass a config.

Reply:
hm, if that can be done at the top level (outside a component), will switch that to that, since react context needs to be made outside the component